server/lib/email.ts

```diff
@@ -248,20 +265,28 @@ const generateEmailFromTemplate = (
   data: SendMessageData = {},
   options: SendMessageOptions = {},
 ) => {
-  const slug = get(options, 'collective.slug') || get(data, 'collective.slug') || 'undefined';
+  const slug = get(options, 'collective.slug') || get(data, 'collective.slug') || '';
   const hostSlug = get(data, 'host.slug');
   const eventSlug = get(data, 'event.slug');
+  const projectSlug = get(data, 'project.slug');
+  const emailId = get(options, 'type') || template;
+
+  // Populate the `listId` and `accountSlug` options to later populate the email headers. We're doing that here for
+  // consistency with the unsubscribe link handling below, but mutating `data` is not ideal and should be refactored.
+  if (options) {
+    options.listId = slug ? `${slug}::${emailId}` : emailId;
+    options.accountSlug = eventSlug ?? projectSlug ?? slug;
+  }
 
   // If we are sending the same email to multiple recipients, it doesn't make sense to allow them to unsubscribe
   if (!isArray(recipient) && !options?.isTransactional) {
-    data.notificationTypeLabel = getNotificationLabel(options.type || template);
-    data.unsubscribeUrl = `${config.host.website}/email/unsubscribe/${encodeURIComponent(
-      recipient || options.bcc,
-    )}/${slug}/${options.type || template}/${generateUnsubscribeToken(
-      recipient || options.bcc,
-      slug,
-      options.type || template,
-    )}`;
+    data.notificationTypeLabel = getNotificationLabel(emailId);
+    if (data.notificationTypeLabel) {
```

Comment:
In all templates, we don't show the unsubscribe link if `notificationTypeLabel` is not set. This change should therefore introduce no regression.